### ISSUE TITLE
abstraction of text decoration

### ISSUE THREAD

**Issue by balmas** [latin-language-toolkit/arethusa#580]
this isn't fully formed in my head yet but I wanted to make a note of it while I'm thinking of it.  It might be interesting to see if we can abstract the colorizer functionality into a more general text decoration capability. This will be particularly useful when we start working with multiple documents.  For example, Marie-Claire would like the ability for metrical attributes on segments in a TEI document to be visible both in the TEI text view and the tree. I'm not sure, but it seems to me that this is analagous in alot of ways to the way the colorizer functionality is currently handled, in that different plugins can define aspects of text decoration, and then the user can choose which to show at any time.

**Comment by LFDM** [comment]
Yes I agree that we want something that compares to colorizers here - there are a couple of open questions with this though:

- Where do we obtain metric information?
  - It is either encoded in a document as information to a whole line/verse
  - But also every word naturally has metric information defined by its lengths and shorts
- Metric analysis could be an own plugin all together. It would define additional token stylings and allow inspection of the meter present, with additional resources about it, with the ability to define the meter etc etc.

There is more, will think about this and then follow-up.

**Comment by balmas** [comment]
@LFDM I agree there is lots to think about here.  And I think it's not limited to meter, this is just one example of the type of data that can come in from a plugin or document that might be used to decorate a token throughout the interface.  As you note, the meter case was interesting because it also cascades down to tokens from a higher-up element in the source document (i.e. the line) -- in the demo code we wanted to decorate just the first word of the line on which the meter was actually encoded in the source document.  Note that code I pushed for this is not at all intended to be something we build on, just a way of demonstrating the type of functionality and providing a way to produce a screenshot :)
